### PR TITLE
feat: CRM search contact/leads on email

### PR DESCRIPTION
**Pull request by asdfryan**
Fixes: SUP-638

Tested locally

### CODE REVIEW

File: packages/core/remotes/impl/hubspot/index.ts [hunk 1]
```diff
@@ -887,40 +867,26 @@ class HubSpotClient extends AbstractCrmRemoteClient implements MarketingAutomati
...
     limit: number,
     after?: string
   ): Promise<RecordsResponseWithFlattenedAssociationsAndTotal> {
-    // Get records
-    // hubspot doesn't set hs_lastmodifieddate for some reason for contact
-    const lastModifiedAtPropertyName = objectType === 'contact' ? 'lastmodifieddate' : 'hs_lastmodifieddate';
     const response = await retryWhenAxiosRateLimited(async () => {
       await this.maybeRefreshAccessToken();
       return await axios.post<HubSpotAPIV3SearchResponse>(
         `${this.baseUrl}/crm/v3/objects/${objectType}/search`,
```
Comment (tomkit) [inline packages/core/remotes/impl/hubspot/index.ts:886]: When do we use axios vs the client?

File: packages/core/remotes/impl/salesforce/index.ts [hunk 2]
```diff
@@ -1379,6 +1418,28 @@ ${modifiedAfter ? `WHERE SystemModstamp > ${modifiedAfter.toISOString()} ORDER B
...
+    const propertiesToFetch = await this.getCommonPropertiesToFetch('lead', fieldMappingConfig);
+    const soql = `SELECT ${propertiesToFetch.join(',')}
+    FROM Lead WHERE Email = '${params.filter.value}'`;
+    const response = await this.#client.query(soql);
+    const records = response.records.map((record) => ({
+      ...fromSalesforceLeadToLead(record),
+      rawData: toMappedProperties(record, fieldMappingConfig),
+    }));
+    return {
+      pagination: {
```
Comment (tomkit) [inline packages/core/remotes/impl/salesforce/index.ts:1434]: Can you add the same comment you did for Hubspot here?

File: packages/core/remotes/impl/salesforce/index.ts [hunk 3]
```diff
@@ -1317,6 +1334,28 @@ ${modifiedAfter ? `WHERE SystemModstamp > ${modifiedAfter.toISOString()} ORDER B
...
+    const propertiesToFetch = await this.getCommonPropertiesToFetch('contact', fieldMappingConfig);
+    const soql = `SELECT ${propertiesToFetch.join(',')}
+    FROM Contact WHERE Email = '${params.filter.value}'`;
+    const response = await this.#client.query(soql);
+    const records = response.records.map((record) => ({
+      ...fromSalesforceContactToContact(record),
+      rawData: toMappedProperties(record, fieldMappingConfig),
+    }));
+    return {
+      pagination: {
```
Comment (tomkit) [inline packages/core/remotes/impl/salesforce/index.ts:1350]: Can you add the same comment you did for Hubspot here?

File: openapi/v2/crm/paths/contacts@search.yaml [hunk 4]
```diff
@@ -0,0 +1,56 @@
+post:
+  operationId: searchContacts
+  summary: Search contacts
+  description: |
+    Search contacts by a filter. Only supported for Salesforce and Hubspot.
```
Comment (tomkit) [inline openapi/v2/crm/paths/contacts@search.yaml:5]: I started adding a support matrix inside of the unified endpoints when we need to to track {provider, dimension} coverage, e.g. https://docs.supaglue.com/api/v2/crm/list-list-memberships

File: openapi/v2/crm/paths/leads@search.yaml [hunk 5]
```diff
@@ -0,0 +1,56 @@
+post:
+  operationId: searchLeads
+  summary: Search leads
+  description: |
+    Search leads by a filter. Only supported for Salesforce.
```
Comment (tomkit) [inline openapi/v2/crm/paths/leads@search.yaml:5]: I started adding a support matrix inside of the unified endpoints when we need to to track {provider, dimension} coverage, e.g. https://docs.supaglue.com/api/v2/crm/list-list-memberships

File: packages/core/remotes/impl/hubspot/index.ts (hunk 1 above)
Comment (asdfryan) [reply packages/core/remotes/impl/hubspot/index.ts:886]: I do not know but I am not changing that logic. Just refactoring this function to be re-usable in multiple places.

**Review by tomkit — APPROVED** [review]

**Pull request merged**